sales_team_security/security/sales_team_security.xml

```diff
@@ -0,0 +1,82 @@
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="sale_order_report_team_rule" model="ir.rule">
+        <field name="name">Sales Team Orders Analysis</field>
+        <field ref="sale.model_sale_report" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="sale_order_line_team_rule" model="ir.rule">
+        <field name="name">Sales Team Order Lines</field>
+        <field ref="sale.model_sale_order_line" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="res_partner_team_rule" model="ir.rule">
+        <field name="name">Sales Team Partners</field>
+        <field ref="base.model_res_partner" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="crm_lead_team_rule" model="ir.rule">
+        <field name="name">Sales Team Leads/Opportunities</field>
+        <field ref="crm.model_crm_lead" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="crm_lead_report_team" model="ir.rule">
+        <field name="name">Sales Team Leads Analysis</field>
+        <field ref="crm.model_crm_lead_report" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="crm_phonecall_team_rule" model="ir.rule">
+        <field name="name">Sales Team Phone Calls</field>
+        <field ref="crm.model_crm_phonecall" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="crm_phonecall_report_team" model="ir.rule">
+        <field name="name">Sales Team Phone Calls Analysis</field>
+        <field ref="crm.model_crm_phonecall_report" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
```

Comment:
I feel this rule and all of the above are just great! They allow you to split your database in a very sensible and powerful way for most salesmen.

However, there's still a problem: What if for any reason I need user from team B see a record from team A? You probably do not want to add him/her to a team just to see a phonecall, partner, etc.

I think all these rules should be expanded with `[('message_follower_ids', 'in', user.partner_id.ids)]` to be able to make any user a follower and s/he gets instant access to that single record.